manager/controllers/app/policy_compiler_interface.go

```diff
@@ -17,12 +17,11 @@ import (
 	"github.com/gdexlab/go-render/render"
 )
 
-func ConstructOpenAPIReq(datasetID string, input *app.FybrikApplication, operation *policymanager.RequestAction) *policymanager.GetPolicyDecisionsRequest {
-	context := make(map[string]interface{}, len(input.Spec.AppInfo))
-	for k, v := range input.Spec.AppInfo {
-		context[k] = v
+func ConstructOpenAPIReq(datasetID string, input *app.FybrikApplication, operation *policymanager.RequestAction) (*policymanager.GetPolicyDecisionsRequest, error) {
+	context, err := utils.StructToMap(&input.Spec.AppInfo)
+	if err != nil {
+		return nil, err
```

Comment:
I'm not sure why a deep copy is needed (in the original source too). Just asking because it seems like a flat copy is enough to match between the types, and also in the future once appinfo becomes a key under context it will even be simpler.

Reply:
ok